### PR TITLE
version and root check

### PR DESCRIPTION
**Pull request by lolpro11**
- Added a check to see if script is run as the root user.
- Added the ability to check and download the latest version.
- Added a check to see if python is already installed.
- Added a check to see if pip is already installed.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by TechStudent11** [comment]
Why do you need to check if the user is root?

**Comment by Firecul** [comment]
To allow installing the dependencies most likely.

**Comment by lolpro11** [comment]
> Why do you need to check if the user is root?

Package managers need root permission.

**Comment by rbeer** [comment]
Like I said in the review, you need to fix the checks.

You are setting `$JQ` and `$PY` when the `command` fails, i.e. when they are *not* installed.
Later, you run the installations when their respective flag is not 0, which is the case when the program is present.

Worse, in line 39 you `purge jq` when it was installed. As it is now, you are removing programs people had already installed.
https://github.com/ThioJoe/YT-Spammer-Purge/pull/326/files#diff-043df5bdbf6639d7a77e1d44c5226fd7371e5259a1e4df3a0dd5d64c30dca44fR39

File: install.sh [hunk 1]
```diff
@@ -1,27 +1,58 @@
...
+		dnf remove jq
+	elif [[ -e /etc/centos-release ]]; then
+		yum remove jq epel-release
+	elif [[ -e /etc/arch-release ]]; then
+		pacman -Rs jq
+	fi
 else
-	echo "Looks like you aren't running this installer on a Debian, Ubuntu, Fedora, CentOS, Arch Linux system."
-	exit 1
+	echo "Did not unistall jq as it was preinstalled before running this script."
```
Comment (Firecul) [inline install.sh:50]: "Did not uninstall jq ..."

File: install_full.sh [hunk 2]
```diff
@@ -1,26 +1,58 @@
...
+		dnf remove jq
+	elif [[ -e /etc/centos-release ]]; then
+		yum remove jq epel-release
+	elif [[ -e /etc/arch-release ]]; then
+		pacman -Rs jq
+	fi
 else
-	echo "Looks like you aren't running this installer on a Debian, Ubuntu, Fedora, CentOS, Arch Linux system."
-	exit 1
+	echo "Did not unistall jq as it was preinstalled before running this script."
```
Comment (Firecul) [inline install_full.sh:50]: "Did not uninstall jq..."

**Comment by TechStudent11** [comment]
@Firecul @lolpro11 Thanks for responding. I totally forgot package managers needed root. But why not just run `sudo` outright?

**Comment by lolpro11** [comment]
> @Firecul @lolpro11 Thanks for responding. I totally forgot package managers needed root. But why not just run `sudo` outright?

I decided to do that, and I fixed the checking.

**Pull request merged**